lib/js/components/DatePickers/DatePickerBox/DatePickerBox.vue

```diff
@@ -0,0 +1,278 @@
+	$self: &;
+
+	border: $border-xs solid transparent;
+	display: flex;
+	flex-direction: column;
+	padding: $space-2xs $space-xs;
+	pointer-events: none;
+	position: relative;
+
+	&.-ds-interactive {
+		background-color: $color-neutral-background-weak;
+		border-color: $color-neutral-background;
+		border-radius: $radius-m;
+		cursor: pointer;
+		pointer-events: all;
+
+		&:hover {
+			background-color: $color-neutral-background-weak-hovered;
+		}
+	}
+
+	&.-ds-disabled {
+		background-color: $color-neutral-background-weak-disabled;
+		pointer-events: none;
+	}
+
+	&.-ds-isOpen {
+		background-color: $color-neutral-background-medium;
+
+		&:hover {
+			background-color: $color-neutral-background-medium;
+		}
+	}
+
+	&__eyebrow {
+		@include info-xs-default-regular;
+	}
+
+	&__dateWrapper {
+		column-gap: $space-4xs;
+		display: flex;
+	}
+
+	&__date {
+		@include text-s-default-regular;
+	}
+
+	&__icon {
+		&.-ds-hiddenOnMobile {
+			display: none;
+
+			@media (#{breakpoint-s()}) {
+				display: flex;
+			}
+		}
+	}
+
+	&__loader {
+		align-items: center;
+		background-color: inherit;
+		border-radius: $radius-m;
+		display: flex;
+		height: calc(100% - $border-s * 2);
+		justify-content: center;
+		left: 0;
+		position: absolute;
+		top: 0;
+		width: calc(100% - $border-s * 2);
+	}
+
+	&.-ds-neutral {
+		@include color-scheme(
+			$color-neutral-text-weak,
+			$color-neutral-text-heavy,
+			$color-neutral-icon,
+			$color-neutral-icon-hovered,
+			$color-neutral-text-heavy,
+			$color-neutral-icon-disabled,
+			$color-neutral-text-disabled,
+			$color-neutral-text-heavy-disabled,
+			$self
+		);
+	}
+
+	&.-ds-warning {
+		@include color-scheme(
+			$color-warning-text,
+			$color-warning-text,
+			$color-warning-icon,
+			$color-warning-icon-hovered,
+			$color-warning-text-hovered,
+			$color-warning-icon-disabled,
+			$color-warning-text-disabled,
+			$color-warning-text-disabled,
+			$self
+		);
+	}
+}
+</style>
+
+<script lang="ts">
+import DsIcon, { ICON_SIZES, ICONS } from '../../Icons/Icon';
+import {
+	DATE_PICKER_COLORS,
+	DATE_PICKER_STATES,
+	DatePickerColors,
+	DatePickerStates,
+} from '../DatePicker/DatePicker.consts';
+import { PropType, toRaw } from 'vue';
+
+export default {
+	name: 'DatePickerBox',
+	components: {
+		DsIcon,
+	},
+	props: {
+		isInteractive: {
+			type: Boolean,
+			default: true,
+		},
+		placeholder: {
+			type: String,
+			default: 'Wybierz datę',
+		},
+		date: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		icon: {
+			type: Object,
+			default: null,
+			validate: (icon) => icon === null || Object.values(ICONS).includes(toRaw(icon)),
+		},
+		isIconHiddenOnMobile: {
+			type: Boolean,
+			default: false,
+		},
+		state: {
+			type: (String as PropType<DatePickerStates>) || null,
+			default: DATE_PICKER_STATES.DEFAULT,
+		},
+		color: {
+			type: String as PropType<DatePickerColors>,
+			default: DATE_PICKER_COLORS.NEUTRAL,
+		},
+		eyebrowText: {
+			type: String,
+			default: '',
```

Comment:
why not `null`? we don't want to change the type? For me, an empty string is completely different from null, null means that there is nothing here - and string '' means that it is simply empty text.

Reply:
It is common pattern in DS. We event did it in Toast